kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -452,11 +452,6 @@ public void proverChecksX(Module specModule, Module mainDefModule) {
                 if (s instanceof Rule && !s.att().contains(Att.SIMPLIFICATION()))
                     errors.add(KEMException.compilerError("Only claims and simplification rules are allowed in proof modules.", s));
             }
-            if (s instanceof Rule && (s.att().contains(Att.SIMPLIFICATION()) && !s.att().contains(Att.ANYWHERE()))) {
-                KLabel kl = m.matchKLabel((Rule) s);
-                if (!m.functions().contains(kl))
-                    errors.add(KEMException.compilerError("Simplification rules need to be function/functional like.", s));
-            }
```

Comment:
Dwight suggested above that I do so. If we kept the check as is, it would fire for every single simplification rule in a spec because they can no longer have the anywhere attribute.